python/meterpreter/ext_server_stdapi.py

```diff
@@ -1777,7 +1777,7 @@ def stdapi_net_config_get_arp_table(request, response):
     result = GetIpNetTable(ipnet_table, size, False)
 
     if result == ERROR_INSUFFICIENT_BUFFER:
-        ipnet_table = ctypes.cast(ctypes.create_string_buffer(b'', size.value), ctypes.c_void_p)
+        ipnet_table = ctypes.cast(ctypes.create_string_buffer(bytes(''), size.value), ctypes.c_void_p)
```

Comment:
This will fail on Python 3 if we don't specify an encoding:

```
 ~/git/metasploit-payloads │ upstream-master:master !1  python                                                                                                                            ✔ │ 3.0.5 Ruby │ 12:22:02 PM 
Python 2.7.18 (default, Jul  1 2022, 10:30:50) 
[GCC 11.2.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> bytes('a')
'a'
>>> b'a'
'a'
>>> exit()
 ~/git/metasploit-payloads │ upstream-master:master !1  python3                                                                                                                     ✔ │ 17s │ 3.0.5 Ruby │ 12:47:03 PM 
Python 3.10.10 (main, Feb  7 2023, 12:19:31) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> b'a'
b'a'
>>> bytes
<class 'bytes'>
>>> bytes('a')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: string argument without an encoding
>>> 
```

Reply:
Unfortunately looks like behavior of this method changed between Python 2 and Python 3:

```
 ~/git/metasploit-payloads │ upstream-master:master !1  python3                                                                                                                     ✔ │ 17s │ 3.0.5 Ruby │ 12:47:03 PM 
Python 3.10.10 (main, Feb  7 2023, 12:19:31) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> b'a'
b'a'
>>> bytes
<class 'bytes'>
>>> bytes('a')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: string argument without an encoding
>>> bytes('')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: string argument without an encoding
>>> bytes('', 'ascii')
b''
>>> bytes('', 'utf-8')
b''
>>> exit
Use exit() or Ctrl-D (i.e. EOF) to exit
>>> exit()
 ~/git/metasploit-payloads │ upstream-master:master !1  python                                                                                                                   ✔ │ 3m 10s │ 3.0.5 Ruby │ 12:50:15 PM 
Python 2.7.18 (default, Jul  1 2022, 10:30:50) 
[GCC 11.2.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> bytes('', 'utf-8')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: str() takes at most 1 argument (2 given)
>>> 
```

Reply:
```suggestion
        if sys.version_info[0] < 3
           ipnet_table = ctypes.cast(ctypes.create_string_buffer(bytes(''), size.value), ctypes.c_void_p)
        else 
           ipnet_table = ctypes.cast(ctypes.create_string_buffer(bytes('', 'utf-8'), size.value), ctypes.c_void_p)
```

Reply:
Upon further review I'm not sure Python 2.5 actually even has a `bytes()` function so I'm not sure if this patch would work on Python 2.5:

```
 ~/git/Python/Python-2.5  ./python                                                                                                                                                   ✔ │ 8s │ 3.0.5 Ruby │ 01:20:41 PM 
Python 2.5 (r25:51908, Feb 16 2023, 13:16:37) 
[GCC 11.3.0] on linux5
Type "help", "copyright", "credits" or "license" for more information.
>>> bytes('')    
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'bytes' is not defined
>>> 
```